whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java

```diff
@@ -106,34 +112,40 @@ public RpslObject apply(final RpslObject rpslObject) {
     }
 
     private boolean isMntnerAuthenticated(final RpslObject rpslObject) {
-        if (CollectionUtils.isEmpty(passwords) && StringUtils.isBlank(token) && (certificates == null || certificates.isEmpty())) {
+        if (CollectionUtils.isEmpty(passwords) && StringUtils.isBlank(token) && (certificates == null || certificates.isEmpty()) && (oAuthSession == null || oAuthSession.getUuid() == null)) {
             return false;
         }
 
+        final List<RpslObject> maintainers = getMaintainers(rpslObject);
+
         final List<RpslAttribute> extendedAuthAttributes = Lists.newArrayList(rpslObject.findAttributes(AttributeType.AUTH));
-        extendedAuthAttributes.addAll(getMntByAuthAttributes(rpslObject));
+        extendedAuthAttributes.addAll(getMntByAuthAttributes(maintainers));
```

Comment:
Shouldn't be better to generate another list with mntners? I mean you are affecting rest of authentication methods adding mnt-by attributes
Shouldn't be any side effect because mnt-by format doesn't match the regex in each of them...however I think that it is safer to manage a different list ad-hoc for apikeyAuthenticated
something like:
`
final List<RpslAttribute> extendedAuthAttributes = Lists.newArrayList(rpslObject.findAttributes(AttributeType.AUTH));
`
`
final List<RpslAttribute> apikeyExtendedAttributes =  extendedAuthAttributes.addAll(getMntByAuthAttributes(maintainers));
`
`
apiKeyAuthenticated(rpslObject, maintainers, apikeyExtendedAttributes)
`